File: market-domain/src/main/java/com/example/market/domain/inspection/scheduling/AppointmentId.java
package com.example.market.domain.inspection.scheduling;

import java.util.Objects;
import java.util.UUID;

public record AppointmentId(UUID value) {
    public AppointmentId { Objects.requireNonNull(value); }
    public static AppointmentId newId() { return new AppointmentId(UUID.randomUUID()); }
    public static AppointmentId of(String s) { return new AppointmentId(UUID.fromString(s)); }
    @Override public String toString() { return value.toString(); }
}

File: market-domain/src/main/kotlin/com/example/market/domain/inspection/scheduling/AppointmentId.kt
package com.example.market.domain.inspection.scheduling

import java.util.UUID

data class AppointmentId(@get:JvmName("value") val value: UUID) {

    override fun toString(): String = value.toString()

    companion object {
        @JvmStatic
        fun newId(): AppointmentId = AppointmentId(UUID.randomUUID())

        @JvmStatic
        fun of(s: String): AppointmentId = AppointmentId(UUID.fromString(s))
    }
}
